src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager.cs

```diff
@@ -0,0 +1,216 @@
+                }
+            }
+        }
+
+        if(_isFluentWindowsThemeEnabled)
+        {
+            _currentApplicationTheme = GetSystemTheme();
+            DwmColorization.UpdateAccentColors();
+        }
+    }
+
+    #endregion
+
+    #region Internal Methods
+
+    internal static void ApplySystemTheme(bool forceUpdate = false)
+    {
+        string systemTheme = GetSystemTheme();
+        bool useLightMode = GetUseLightTheme();
+        Color systemAccentColor = DwmColorization.GetSystemAccentColor();
+
+        var windows = Application.Current?.Windows;
+        if(windows != null)
+        {
+            ApplyTheme(windows , systemTheme, useLightMode, systemAccentColor, forceUpdate);
+        }
+    }
+
+    internal static void ApplySystemTheme(Window window, bool forceUpdate = false)
+    {
+        string systemTheme = GetSystemTheme();
+        bool useLightMode = GetUseLightTheme();
+        Color systemAccentColor = DwmColorization.GetSystemAccentColor();
+        ApplyTheme(new ArrayList { window } , systemTheme, useLightMode, systemAccentColor, forceUpdate);
+    }
+
+    internal static void ApplyTheme(
+        IEnumerable windows, 
+        string requestedTheme, 
+        bool requestedUseLightMode,
+        Color requestedAccentColor, 
+        bool forceUpdate = false)
+    {
+        bool needsUpdate = forceUpdate;
+        
+        if(requestedUseLightMode != _currentUseLightMode || DwmColorization.GetSystemAccentColor() != DwmColorization.CurrentApplicationAccentColor)
+        {
+            DwmColorization.UpdateAccentColors();
+            needsUpdate = true;
+        }
+
+        if(needsUpdate || requestedTheme != _currentApplicationTheme || requestedUseLightMode != _currentUseLightMode)
+        {
+            Uri dictionaryUri = GetFluentWindowThemeResourceUri(requestedTheme, requestedUseLightMode);
+
+            UpdateFluentWindowsThemeResources(dictionaryUri);
+
+            foreach(Window window in windows)
+            {
+                SetImmersiveDarkMode(window, !requestedUseLightMode);
+                window.CoerceValue(Window.WindowBackdropTypeProperty);
+            }
+
+            _currentApplicationTheme = requestedTheme;
+            _currentUseLightMode = requestedUseLightMode;
+        }
+    }
+
+    private static bool SetImmersiveDarkMode(Window window, bool useDarkMode)
+    {
+        if (window == null)
+        {
+            return false;
+        }
+
+        IntPtr handle = new WindowInteropHelper(window).Handle;
+
+        if (handle != IntPtr.Zero)
+        {
+            var dwmResult = NativeMethods.DwmSetWindowAttributeUseImmersiveDarkMode(handle, useDarkMode);
+            return dwmResult == HRESULT.S_OK;
+        }
+
+        return false;
+    }
+
+    #region Helper Methods
+
+    internal static string GetSystemTheme()
+    {
+        string systemTheme = Registry.GetValue(_regThemeKeyPath,
+            "CurrentTheme", "aero.theme") as string ?? String.Empty;
```

Comment:
This returns "aero.theme" for one set of failures but `string.Empty` (you should use lowercase `string`) for other set of failures. Shouldn't therebe the same outcome? Pass null for the default value and then do ?? "aero.theme"